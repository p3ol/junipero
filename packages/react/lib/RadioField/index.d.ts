import React, { MutableRefObject } from 'react';

export declare type RadioFieldRef = {
  optionRefs: Array<MutableRefObject<any>>;
  inputRefs: Array<MutableRefObject<any>>;
  innterRef: MutableRefObject<any>;
  dirty: Boolean;
  value: any;
  isJunipero: Boolean;
  valid: Boolean;
};

declare interface RadioFieldProps extends React.ComponentPropsWithRef<any> {
  disabled?: Boolean;
  required?: Boolean;
  valid?: Boolean;
  options?: Array<any>;
  className?: String;
  name?: String;
  value?: any;
  onChange?: (props: { value: any; valid: Boolean }) => void;
  onValidate?: (
    value: any,
    options: { dirty?: Boolean; required?: Boolean }
  ) => Boolean;
  parseValue?: (option: any) => any;
  parseTitle?: (option: any) => String;
  parseDescription?: (option: any) => String;
  ref?: MutableRefObject<RadioFieldRef | undefined>;
}

declare function RadioField(props: RadioFieldProps): JSX.Element;

export default RadioField;
