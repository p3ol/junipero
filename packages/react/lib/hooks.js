import { useContext } from 'react';

import { FieldControlContext } from './contexts';

export const useFieldControl = () => useContext(FieldControlContext);
