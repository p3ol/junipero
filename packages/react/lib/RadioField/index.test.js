import { act, render, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import RadioField from '.';

describe('<RadioField />', () => {
  it('should render', () => {
    const { unmount, container } = render(<RadioField />);
    expect(container).toMatchSnapshot();

    unmount();
  });

  it('should render as many options as provided', () => {
    const { unmount, container } = render(
      <RadioField
        options={[1, 2, 3, 5]}
      />
    );
    expect(container).toMatchSnapshot();

    unmount();
  });

  it('should check if provided value is in options', () => {
    const { unmount, container } = render(
      <RadioField
        options={[1, 2]}
        value={1}
      />
    );
    expect(container).toMatchSnapshot();

    unmount();
  });

  it('should toggle check if a radio is clicked', async () => {
    const onChangeMock = jest.fn();
    const user = userEvent.setup();
    const { unmount, container } = render(
      <RadioField
        options={[1, 2]}
        value={1}
        onChange={onChangeMock}
      />
    );
    expect(container).toMatchSnapshot('option 1 checked');
    const input = container.querySelectorAll('input')[1];

    user.click(input);

    await waitFor(() =>
      expect(onChangeMock).toHaveBeenCalledWith({ valid: true, value: 2 })
    );

    expect(container).toMatchSnapshot('option 2 checked');
    unmount();
  });

  it('should trigger onChange when user hit enter ' +
  'and radio is focused', async () => {
    const onChangeMock = jest.fn();
    const user = userEvent.setup();
    const { unmount, container } = render(
      <RadioField
        options={[1, 2]}
        value={1}
        onChange={onChangeMock}
      />
    );
    expect(container).toMatchSnapshot('option 1 checked');

    await act(async () => {
      container.querySelectorAll('label')[1].focus();
    });
    await user.keyboard('{Enter}');

    await waitFor(() =>
      expect(onChangeMock).toHaveBeenCalledWith({ valid: true, value: 2 })
    );

    expect(container).toMatchSnapshot('option 2 checked');
    unmount();
  });

  it('should trigger onChange when user hit space ' +
  'and radio is focused', async () => {
    const onChangeMock = jest.fn();
    const user = userEvent.setup();
    const { unmount, container } = render(
      <RadioField
        options={[1, 2]}
        value={1}
        onChange={onChangeMock}
      />
    );
    expect(container).toMatchSnapshot('option 1 checked');

    await act(async () => {
      container.querySelectorAll('label')[1].focus();
    });
    await user.keyboard('{ }');

    await waitFor(() =>
      expect(onChangeMock).toHaveBeenCalledWith({ valid: true, value: 2 })
    );

    expect(container).toMatchSnapshot('option 2 checked');
    unmount();
  });

  it('should not trigger onChange if ' +
  'a radio is clicked but its disabled', async () => {
    const onChangeMock = jest.fn();
    const user = userEvent.setup();
    const { unmount, container } = render(
      <RadioField
        options={[1, 2]}
        value={1}
        disabled={true}
        onChange={onChangeMock}
      />
    );
    expect(container).toMatchSnapshot('option 1 checked');
    const input = container.querySelectorAll('input')[1];

    user.click(input);

    await waitFor(() =>
      expect(onChangeMock).not.toHaveBeenCalled()
    );

    expect(container).toMatchSnapshot('option 1 checked again');
    unmount();
  });

  it('should not trigger onChange if ' +
  'a radio if just this option is disabled', async () => {
    const onChangeMock = jest.fn();
    const user = userEvent.setup();
    const { unmount, container } = render(
      <RadioField
        options={[{ value: 'peach' }, { value: 'Orange', disabled: true }]}
        value="peach"
        onChange={onChangeMock}
      />
    );
    expect(container).toMatchSnapshot('option 1 checked');
    const input = container.querySelectorAll('input')[1];

    user.click(input);

    expect(onChangeMock).not.toHaveBeenCalled();

    expect(container).toMatchSnapshot('option 1 checked again');
    unmount();
  });

  it('should add box if boxed class has been passed to Radio', () => {
    const { unmount, container } = render(
      <RadioField
        className="boxed"
        options={[1, 2]}
        value={1}
      />
    );
    expect(container).toMatchSnapshot();

    unmount();
  });

  it('should set error state if onValidate function fails', async () => {
    const onChangeMock = jest.fn();
    const user = userEvent.setup();
    const onValidateMock = jest.fn().mockImplementation(
      (val, { dirty }) => val !== 'orange' || !dirty
    );
    const { unmount, container } = render(
      <RadioField
        options={[{ value: 'peach' }, { value: 'orange' }]}
        value="peach"
        onChange={onChangeMock}
        onValidate={onValidateMock}
      />
    );
    expect(container).toMatchSnapshot('option 1 checked');
    const orangeInput = container.querySelectorAll('input')[1];
    user.click(orangeInput);
    await waitFor(() => expect(onValidateMock).toHaveBeenCalledWith(
      'orange', { dirty: true, required: false }
    ));
    expect(container).toMatchSnapshot('invalid');
    const peachInput = container.querySelectorAll('input')[0];
    user.click(peachInput);
    await waitFor(() => expect(onValidateMock).toHaveBeenCalledTimes(2));

    expect(container).toMatchSnapshot('valid');
    unmount();
  });

  it('should set error state if field is required with ' +
  'default validate function', async () => {
    const user = userEvent.setup();
    const onChangeMock = jest.fn();
    const options = [1];
    const { unmount, container, debug } = render(
      <RadioField
        parseValue={() => null}
        options={options}
        required={true}
        onChange={onChangeMock}
      />
    );
    debug();
    user.click(container.querySelector('input'));

    await waitFor(() => expect(onChangeMock).toHaveBeenCalled());
    expect(container).toMatchSnapshot('invalid');
    unmount();
  });

  it('should not set error state if field is not required with ' +
  'default validate function', async () => {
    const user = userEvent.setup();
    const onChangeMock = jest.fn();
    const options = [1];
    const { unmount, container, debug } = render(
      <RadioField
        parseValue={() => null}
        options={options}
        required={false}
        onChange={onChangeMock}
      />
    );
    debug();
    user.click(container.querySelector('input'));

    await waitFor(() => expect(onChangeMock).toHaveBeenCalled());
    expect(container).toMatchSnapshot('valid');
    unmount();
  });

  it('should use custom onValidate function if provided', async () => {
    const mockOnValidate = jest.fn().mockImplementation(() => true);
    const user = userEvent.setup();

    const options = [{
      value: 'peach',
      title: 'Delicious peach',
      description: 'this is a delicious peach',
    }];

    const { unmount, container } = render(
      <RadioField
        onValidate={mockOnValidate}
        options={options}
      />
    );

    const peachInput = container.querySelectorAll('input')[0];
    user.click(peachInput);
    await waitFor(() =>
      expect(mockOnValidate).toHaveBeenCalledWith(
        'peach', { dirty: true, required: false }
      )
    );
    unmount();
  });

  it('should use given parse title, description and value if ' +
  'function are provided', async () => {
    const options = [{
      cutomValue: 'peach',
      customTitle: 'Delicious peach',
      customDescription: 'this is a delicious peach',
    }];
    const mockParseTitle = jest.fn().mockImplementation(
      option => option.customTitle
    );
    const mockParseValue = jest.fn().mockImplementation(
      option => option.cutomValue
    );
    const mockParseDescription = jest.fn().mockImplementation(
      option => option.customDescription
    );
    const { unmount, container } = render(
      <RadioField
        parseTitle={mockParseTitle}
        parseValue={mockParseValue}
        parseDescription={mockParseDescription}
        options={options}
        value="peach"
      />
    );

    expect(mockParseTitle).toHaveBeenCalledWith(options[0]);
    expect(mockParseValue).toHaveBeenCalledWith(options[0]);
    expect(mockParseDescription).toHaveBeenCalledWith(options[0]);

    expect(container).toMatchSnapshot();
    unmount();
  });

  it('should not display description if ' +
  'parseDescription return empty string', () => {
    const options = [{
      cutomValue: 'peach',
      customTitle: 'Delicious peach',
      customDescription: 'this is a delicious peach',
    }];
    const mockParseDescription = jest.fn().mockImplementation(
      () => ''
    );

    const { unmount, container } = render(
      <RadioField
        parseDescription={mockParseDescription}
        options={options}
        value="peach"
      />
    );
    expect(mockParseDescription).toHaveBeenCalledWith(options[0]);
    expect(container).toMatchSnapshot();
    unmount();
  });

});
