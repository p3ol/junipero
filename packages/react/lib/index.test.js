import { useReducer } from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import { mockState } from '@junipero/core';
import { useTimeout } from '@junipero/hooks';

import TextField from './TextField';

describe('mockState: inside react', () => {
  it('should fail update the state correctly and return the old ' +
    'state', async () => {
    const Comp = () => {
      const [state, dispatch] = useReducer(mockState, {
        value: {
          text: 'This is a test',
          content: [],
        },
      });

      useTimeout(() => {
        dispatch({ value: { ...state.value, content: ['foo'] } });
      }, 1, []);

      const onTextChange = ({ value }) => {
        dispatch({ value: { ...state.value, text: value } });
      };

      const onContentChange = () => {
        dispatch({
          value: { ...state.value, content: [...state.value.content, 'bar'] },
        });
      };

      return (
        <div>
          <TextField
            data-testid="Field"
            value={state.value.text}
            onChange={onTextChange}
          />
          <button onClick={onContentChange}>Add content</button>
          <div>Text: { state.value.text }</div>
          <div>Content: { state.value.content.join(', ') }</div>
        </div>
      );
    };

    const { container, unmount } = render(<Comp />);

    fireEvent.change(screen.getByTestId('Field'),
      { target: { value: 'This is a test 2' } });

    await waitFor(() => screen.getByText('Content: foo'));
    expect(container).toMatchSnapshot('With updated text field value');

    fireEvent.click(screen.getByText('Add content'));

    expect(container)
      .toMatchSnapshot('With updated content but with old text value');

    unmount();
  });

  it('should not fail update the state correctly and return the new ' +
    'state when using mockState with a callback', async () => {
    const Comp = () => {
      const [state, dispatch] = useReducer(mockState, {
        value: {
          text: 'This is a test',
          content: [],
        },
      });

      useTimeout(() => {
        dispatch(s => ({ value: { ...s.value, content: ['foo'] } }));
      }, 1, []);

      const onTextChange = ({ value }) => {
        dispatch(s => ({ value: { ...s.value, text: value } }));
      };

      const onContentChange = () => {
        dispatch(s => ({
          value: { ...s.value, content: [...s.value.content, 'bar'] },
        }));
      };

      return (
        <div>
          <TextField
            data-testid="Field"
            value={state.value.text}
            onChange={onTextChange}
          />
          <button onClick={onContentChange}>Add content</button>
          <div>Text: { state.value.text }</div>
          <div>Content: { state.value.content.join(', ') }</div>
        </div>
      );
    };

    const { container, unmount } = render(<Comp />);

    fireEvent.change(screen.getByTestId('Field'),
      { target: { value: 'This is a test 2' } });

    await waitFor(() => screen.getByText('Content: foo'));
    expect(container).toMatchSnapshot('With updated text field value');

    fireEvent.click(screen.getByText('Add content'));

    expect(container)
      .toMatchSnapshot('With updated content AND updated text field valud');

    unmount();
  });
});
