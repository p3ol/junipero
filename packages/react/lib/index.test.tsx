import { useReducer } from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import { mockState } from '@junipero/core';
import { useTimeout } from '@junipero/hooks';

import type { FieldContent } from './types';
import TextField from './TextField';

interface State {
  value: {
    text: string;
    content: string[];
  };
}

describe('mockState: inside react', () => {
  it('should correctly update the state and return the new ' +
    'state when using mockState with a callback', async () => {
    const Comp = () => {
      const [state, dispatch] = useReducer(mockState<State>, {
        value: {
          text: 'This is a test',
          content: [],
        },
      });

      useTimeout(() => {
        dispatch(s => ({ value: { ...s.value, content: ['foo'] } }));
      }, 1, []);

      const onTextChange = ({ value }: FieldContent) => {
        dispatch(s => ({ value: { ...s.value, text: value } }));
      };

      const onContentChange = () => {
        dispatch(s => ({
          value: { ...s.value, content: [...s.value.content, 'bar'] },
        }));
      };

      return (
        <div>
          <TextField
            data-testid="Field"
            value={state.value.text}
            onChange={onTextChange}
          />
          <button onClick={onContentChange}>Add content</button>
          <div>Text: { state.value.text }</div>
          <div>Content: { state.value.content.join(', ') }</div>
        </div>
      );
    };

    const { container, unmount } = render(<Comp />);

    fireEvent.change(screen.getByTestId('Field'),
      { target: { value: 'This is a test 2' } });

    await waitFor(() => screen.getByText('Content: foo'));
    expect(container).toMatchSnapshot('With updated text field value');

    fireEvent.click(screen.getByText('Add content'));

    expect(container)
      .toMatchSnapshot('With updated content AND updated text field valud');

    unmount();
  });
});
