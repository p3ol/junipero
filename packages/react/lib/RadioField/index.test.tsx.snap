// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`<RadioField /> should add box if boxed class has been passed to Radio 1`] = `
<div>
  <div
    class="junipero radio-field pristine valid boxed"
    role="radiogroup"
  >
    <label
      class="checked"
      role="radio"
      tabindex="0"
    >
      <input
        checked=""
        type="radio"
        value="1"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          1
        </div>
      </div>
    </label>
    <label
      class=""
      role="radio"
      tabindex="-1"
    >
      <input
        type="radio"
        value="2"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          2
        </div>
      </div>
    </label>
  </div>
</div>
`;

exports[`<RadioField /> should change options if options props change accross the time: firm options 1`] = `
<div>
  <div
    class="junipero radio-field pristine valid"
    role="radiogroup"
  >
    <label
      class=""
      role="radio"
      tabindex="0"
    >
      <input
        type="radio"
        value="facebook"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          Facebook
        </div>
      </div>
    </label>
    <label
      class=""
      role="radio"
      tabindex="-1"
    >
      <input
        type="radio"
        value="microsoft"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          Microsoft
        </div>
      </div>
    </label>
    <label
      class=""
      role="radio"
      tabindex="-1"
    >
      <input
        type="radio"
        value="apple"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          Apple
        </div>
      </div>
    </label>
    <label
      class=""
      role="radio"
      tabindex="-1"
    >
      <input
        type="radio"
        value="google"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          Google
        </div>
      </div>
    </label>
  </div>
</div>
`;

exports[`<RadioField /> should change options if options props change accross the time: fruits options 1`] = `
<div>
  <div
    class="junipero radio-field pristine valid"
    role="radiogroup"
  >
    <label
      class=""
      role="radio"
      tabindex="0"
    >
      <input
        type="radio"
        value="peach"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          Peach
        </div>
      </div>
    </label>
    <label
      class=""
      role="radio"
      tabindex="-1"
    >
      <input
        type="radio"
        value="orange"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          Orange
        </div>
      </div>
    </label>
    <label
      class=""
      role="radio"
      tabindex="-1"
    >
      <input
        type="radio"
        value="banana"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          Banana
        </div>
      </div>
    </label>
  </div>
</div>
`;

exports[`<RadioField /> should change value if value prop change accross the time: orange checked 1`] = `
<div>
  <div
    class="junipero radio-field pristine valid"
    role="radiogroup"
  >
    <label
      class=""
      role="radio"
      tabindex="-1"
    >
      <input
        checked=""
        type="radio"
        value="peach"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          Peach
        </div>
      </div>
    </label>
    <label
      class="checked"
      role="radio"
      tabindex="0"
    >
      <input
        type="radio"
        value="orange"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          Orange
        </div>
      </div>
    </label>
    <label
      class=""
      role="radio"
      tabindex="-1"
    >
      <input
        type="radio"
        value="banana"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          Banana
        </div>
      </div>
    </label>
  </div>
</div>
`;

exports[`<RadioField /> should change value if value prop change accross the time: peach checked 1`] = `
<div>
  <div
    class="junipero radio-field pristine valid"
    role="radiogroup"
  >
    <label
      class="checked"
      role="radio"
      tabindex="0"
    >
      <input
        checked=""
        type="radio"
        value="peach"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          Peach
        </div>
      </div>
    </label>
    <label
      class=""
      role="radio"
      tabindex="-1"
    >
      <input
        type="radio"
        value="orange"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          Orange
        </div>
      </div>
    </label>
    <label
      class=""
      role="radio"
      tabindex="-1"
    >
      <input
        type="radio"
        value="banana"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          Banana
        </div>
      </div>
    </label>
  </div>
</div>
`;

exports[`<RadioField /> should check if provided value is in options 1`] = `
<div>
  <div
    class="junipero radio-field pristine valid"
    role="radiogroup"
  >
    <label
      class="checked"
      role="radio"
      tabindex="0"
    >
      <input
        checked=""
        type="radio"
        value="1"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          1
        </div>
      </div>
    </label>
    <label
      class=""
      role="radio"
      tabindex="-1"
    >
      <input
        type="radio"
        value="2"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          2
        </div>
      </div>
    </label>
  </div>
</div>
`;

exports[`<RadioField /> should navigate and toggle with keyboard: option 1 checked 1`] = `
<div>
  <div
    class="junipero radio-field pristine valid"
    role="radiogroup"
  >
    <label
      class=""
      role="radio"
      tabindex="0"
    >
      <input
        type="radio"
        value="1"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          1
        </div>
      </div>
    </label>
    <label
      class=""
      role="radio"
      tabindex="-1"
    >
      <input
        type="radio"
        value="2"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          2
        </div>
      </div>
    </label>
    <label
      class=""
      role="radio"
      tabindex="-1"
    >
      <input
        type="radio"
        value="3"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          3
        </div>
      </div>
    </label>
  </div>
</div>
`;

exports[`<RadioField /> should navigate and toggle with keyboard: option 2 checked 1`] = `
<div>
  <div
    class="junipero radio-field dirty valid"
    role="radiogroup"
  >
    <label
      class=""
      role="radio"
      tabindex="-1"
    >
      <input
        type="radio"
        value="1"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          1
        </div>
      </div>
    </label>
    <label
      class="checked"
      role="radio"
      tabindex="0"
    >
      <input
        type="radio"
        value="2"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          2
        </div>
      </div>
    </label>
    <label
      class=""
      role="radio"
      tabindex="-1"
    >
      <input
        type="radio"
        value="3"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          3
        </div>
      </div>
    </label>
  </div>
</div>
`;

exports[`<RadioField /> should not display description if parseDescription return empty string 1`] = `
<div>
  <div
    class="junipero radio-field pristine valid"
    role="radiogroup"
  >
    <label
      class=""
      role="radio"
      tabindex="-1"
    >
      <input
        type="radio"
        value="[object Object]"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          [object Object]
        </div>
      </div>
    </label>
  </div>
</div>
`;

exports[`<RadioField /> should not trigger onChange if a radio if just this option is disabled: option 1 checked 1`] = `
<div>
  <div
    class="junipero radio-field pristine valid"
    role="radiogroup"
  >
    <label
      class="checked"
      role="radio"
      tabindex="0"
    >
      <input
        checked=""
        type="radio"
        value="peach"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          [object Object]
        </div>
      </div>
    </label>
    <label
      class="disabled"
      role="radio"
      tabindex="-1"
    >
      <input
        type="radio"
        value="Orange"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          [object Object]
        </div>
      </div>
    </label>
  </div>
</div>
`;

exports[`<RadioField /> should not trigger onChange if a radio if just this option is disabled: option 1 checked again 1`] = `
<div>
  <div
    class="junipero radio-field pristine valid"
    role="radiogroup"
  >
    <label
      class="checked"
      role="radio"
      tabindex="0"
    >
      <input
        checked=""
        type="radio"
        value="peach"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          [object Object]
        </div>
      </div>
    </label>
    <label
      class="disabled"
      role="radio"
      tabindex="-1"
    >
      <input
        type="radio"
        value="Orange"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          [object Object]
        </div>
      </div>
    </label>
  </div>
</div>
`;

exports[`<RadioField /> should not trigger onChange if a radio is clicked but its disabled: option 1 checked 1`] = `
<div>
  <div
    class="junipero radio-field disabled pristine valid"
    role="radiogroup"
  >
    <label
      class="checked disabled"
      role="radio"
      tabindex="0"
    >
      <input
        checked=""
        type="radio"
        value="1"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          1
        </div>
      </div>
    </label>
    <label
      class="disabled"
      role="radio"
      tabindex="-1"
    >
      <input
        type="radio"
        value="2"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          2
        </div>
      </div>
    </label>
  </div>
</div>
`;

exports[`<RadioField /> should not trigger onChange if a radio is clicked but its disabled: option 1 checked again 1`] = `
<div>
  <div
    class="junipero radio-field disabled pristine valid"
    role="radiogroup"
  >
    <label
      class="checked disabled"
      role="radio"
      tabindex="0"
    >
      <input
        checked=""
        type="radio"
        value="1"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          1
        </div>
      </div>
    </label>
    <label
      class="disabled"
      role="radio"
      tabindex="-1"
    >
      <input
        type="radio"
        value="2"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          2
        </div>
      </div>
    </label>
  </div>
</div>
`;

exports[`<RadioField /> should render 1`] = `
<div>
  <div
    class="junipero radio-field pristine valid"
    role="radiogroup"
  />
</div>
`;

exports[`<RadioField /> should render as many options as provided 1`] = `
<div>
  <div
    class="junipero radio-field pristine valid"
    role="radiogroup"
  >
    <label
      class=""
      role="radio"
      tabindex="0"
    >
      <input
        type="radio"
        value="1"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          1
        </div>
      </div>
    </label>
    <label
      class=""
      role="radio"
      tabindex="-1"
    >
      <input
        type="radio"
        value="2"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          2
        </div>
      </div>
    </label>
    <label
      class=""
      role="radio"
      tabindex="-1"
    >
      <input
        type="radio"
        value="3"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          3
        </div>
      </div>
    </label>
    <label
      class=""
      role="radio"
      tabindex="-1"
    >
      <input
        type="radio"
        value="5"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          5
        </div>
      </div>
    </label>
  </div>
</div>
`;

exports[`<RadioField /> should set error state if onValidate function fails: invalid 1`] = `
<div>
  <div
    class="junipero radio-field dirty invalid"
    role="radiogroup"
  >
    <label
      class=""
      role="radio"
      tabindex="-1"
    >
      <input
        checked=""
        type="radio"
        value="peach"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          [object Object]
        </div>
      </div>
    </label>
    <label
      class="checked"
      role="radio"
      tabindex="0"
    >
      <input
        type="radio"
        value="orange"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          [object Object]
        </div>
      </div>
    </label>
  </div>
</div>
`;

exports[`<RadioField /> should set error state if onValidate function fails: option 1 checked 1`] = `
<div>
  <div
    class="junipero radio-field pristine valid"
    role="radiogroup"
  >
    <label
      class="checked"
      role="radio"
      tabindex="0"
    >
      <input
        checked=""
        type="radio"
        value="peach"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          [object Object]
        </div>
      </div>
    </label>
    <label
      class=""
      role="radio"
      tabindex="-1"
    >
      <input
        type="radio"
        value="orange"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          [object Object]
        </div>
      </div>
    </label>
  </div>
</div>
`;

exports[`<RadioField /> should set error state if onValidate function fails: valid 1`] = `
<div>
  <div
    class="junipero radio-field dirty valid"
    role="radiogroup"
  >
    <label
      class="checked"
      role="radio"
      tabindex="0"
    >
      <input
        checked=""
        type="radio"
        value="peach"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          [object Object]
        </div>
      </div>
    </label>
    <label
      class=""
      role="radio"
      tabindex="-1"
    >
      <input
        type="radio"
        value="orange"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          [object Object]
        </div>
      </div>
    </label>
  </div>
</div>
`;

exports[`<RadioField /> should toggle check if a radio is clicked: option 1 checked 1`] = `
<div>
  <div
    class="junipero radio-field pristine valid"
    role="radiogroup"
  >
    <label
      class="checked"
      role="radio"
      tabindex="0"
    >
      <input
        checked=""
        type="radio"
        value="1"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          1
        </div>
      </div>
    </label>
    <label
      class=""
      role="radio"
      tabindex="-1"
    >
      <input
        type="radio"
        value="2"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          2
        </div>
      </div>
    </label>
  </div>
</div>
`;

exports[`<RadioField /> should toggle check if a radio is clicked: option 2 checked 1`] = `
<div>
  <div
    class="junipero radio-field dirty valid"
    role="radiogroup"
  >
    <label
      class=""
      role="radio"
      tabindex="-1"
    >
      <input
        checked=""
        type="radio"
        value="1"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          1
        </div>
      </div>
    </label>
    <label
      class="checked"
      role="radio"
      tabindex="0"
    >
      <input
        type="radio"
        value="2"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          2
        </div>
      </div>
    </label>
  </div>
</div>
`;

exports[`<RadioField /> should trigger onChange when user hit enter and radio is focused: option 1 checked 1`] = `
<div>
  <div
    class="junipero radio-field pristine valid"
    role="radiogroup"
  >
    <label
      class="checked"
      role="radio"
      tabindex="0"
    >
      <input
        checked=""
        type="radio"
        value="1"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          1
        </div>
      </div>
    </label>
    <label
      class=""
      role="radio"
      tabindex="-1"
    >
      <input
        type="radio"
        value="2"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          2
        </div>
      </div>
    </label>
  </div>
</div>
`;

exports[`<RadioField /> should trigger onChange when user hit enter and radio is focused: option 2 checked 1`] = `
<div>
  <div
    class="junipero radio-field dirty valid"
    role="radiogroup"
  >
    <label
      class=""
      role="radio"
      tabindex="-1"
    >
      <input
        checked=""
        type="radio"
        value="1"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          1
        </div>
      </div>
    </label>
    <label
      class="checked"
      role="radio"
      tabindex="0"
    >
      <input
        type="radio"
        value="2"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          2
        </div>
      </div>
    </label>
  </div>
</div>
`;

exports[`<RadioField /> should trigger onChange when user hit space and radio is focused: option 1 checked 1`] = `
<div>
  <div
    class="junipero radio-field pristine valid"
    role="radiogroup"
  >
    <label
      class="checked"
      role="radio"
      tabindex="0"
    >
      <input
        checked=""
        type="radio"
        value="1"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          1
        </div>
      </div>
    </label>
    <label
      class=""
      role="radio"
      tabindex="-1"
    >
      <input
        type="radio"
        value="2"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          2
        </div>
      </div>
    </label>
  </div>
</div>
`;

exports[`<RadioField /> should trigger onChange when user hit space and radio is focused: option 2 checked 1`] = `
<div>
  <div
    class="junipero radio-field dirty valid"
    role="radiogroup"
  >
    <label
      class=""
      role="radio"
      tabindex="-1"
    >
      <input
        checked=""
        type="radio"
        value="1"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          1
        </div>
      </div>
    </label>
    <label
      class="checked"
      role="radio"
      tabindex="0"
    >
      <input
        type="radio"
        value="2"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          2
        </div>
      </div>
    </label>
  </div>
</div>
`;

exports[`<RadioField /> should use given parse title, description and value if function are provided 1`] = `
<div>
  <div
    class="junipero radio-field pristine valid"
    role="radiogroup"
  >
    <label
      class="checked"
      role="radio"
      tabindex="0"
    >
      <input
        checked=""
        type="radio"
        value="peach"
      />
      <div
        class="inner"
      />
      <div
        class="label"
      >
        <div
          class="title"
        >
          Delicious peach
        </div>
        <div
          class="description"
        >
          this is a delicious peach
        </div>
      </div>
    </label>
  </div>
</div>
`;
